Lib/test/test_ntpath.py

```diff
@@ -270,6 +277,13 @@ def test_normpath(self):
         tester("ntpath.normpath('//server/share/../..')",  '\\\\server\\share\\')
         tester("ntpath.normpath('//server/share/../../')", '\\\\server\\share\\')
 
+        # gh-96290: don't normalize partial/invalid UNC drives
+        tester("ntpath.normpath('\\\\foo\\bar')", '\\\\foo\\bar')
+        tester("ntpath.normpath('\\\\foo\\\\')", '\\\\foo\\\\')
```

Comment:
With the suggested change to `splitdrive()`, the result should be `'\\\\foo\\'`. However, the C implementation of `nt._path_normpath()` doesn't implement the API behavior for an empty share. This case wouldn't be testable until that's updated. (The C implementation still lacks support for "UNC" device drives as well.)